pandas/core/groupby/grouper.py

```diff
@@ -563,8 +563,16 @@ def indices(self):
         if isinstance(self.grouper, ops.BaseGrouper):
             return self.grouper.indices
 
-        values = Categorical(self.grouper)
-        return values._reverse_indexer()
+        if self.sort:
+            values = Categorical(self.grouper)
+            return values._reverse_indexer()
+        else:
+            # GH 36889
+            indices_dict = {}
```

Comment:
hmm i think _reverse_indexer can do this right?

Reply:
`Categorical._reverse_indexer()` sorts automatically. The `Categorical` constructor alone auto-sorts

Reply:
sure, but you can factorize and pass sort then do this in both cases.

Reply:
So use `factorize` in both cases instead of going through `Categorical._reverse_indexer()` for `sort=True`?. Sure I can try that

Reply:
yeah just tring to make DRY